code/game/objects/items/storage/boxes/debug_boxes.dm

```diff
@@ -0,0 +1,162 @@
+/obj/item/storage/box/debugbox
+	illustration = null
+
+/obj/item/storage/box/debugbox/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 100
+	atom_storage.max_total_storage = 600
+	atom_storage.max_specific_storage = WEIGHT_CLASS_GIGANTIC
+
+/obj/item/storage/box/debugbox/tools
+	name = "box of debug tools"
+	icon_state = "syndiebox"
+	storage_type = /datum/storage/box/debug
+
+/obj/item/storage/box/debugbox/tools/PopulateContents()
+	var/static/items_inside = list(
+		/obj/item/card/emag=1,
+		/obj/item/construction/rcd/combat/admin=1,
+		/obj/item/disk/tech_disk/debug=1,
+		/obj/item/flashlight/emp/debug=1,
+		/obj/item/geiger_counter=1,
+		/obj/item/healthanalyzer/advanced=1,
+		/obj/item/modular_computer/pda/heads/captain=1,
+		/obj/item/pipe_dispenser=1,
+		/obj/item/stack/spacecash/c1000=50,
+		/obj/item/storage/box/beakers/bluespace=1,
+		/obj/item/storage/box/beakers/variety=1,
+		/obj/item/storage/bag/sheetsnatcher/debug=1,
+		/obj/item/uplink/debug=1,
+		/obj/item/uplink/nuclear/debug=1,
+		/obj/item/clothing/ears/earmuffs/debug=1,
+		/obj/item/gps/visible_debug=1,
+		/obj/item/clothing/glasses/meson/engine/admin=1,
+		/obj/item/storage/box/debugbox/guns=1,
+		)
+	generate_items_inside(items_inside, src)
+
+//---- Box of gun boxes
+/obj/item/storage/box/debugbox/guns
+	name = "box of every gun"
+	icon_state = "gunbox"
+	desc = "One must peer within in order to obtain what they truly desire."
+
+/obj/item/storage/box/debugbox/guns/PopulateContents()
+	for(var/obj/box as anything in subtypesof(/obj/item/storage/box/debugbox/guns))
+		new box(src)
+
+//---- Boxes of ballistics
+/obj/item/storage/box/debugbox/guns/shotgun
+	name = "box of shotguns"
+	icon_state = "shotgunbox"
+	desc = "Holds a lot of shotguns"
+
+/obj/item/storage/box/debugbox/guns/shotgun/PopulateContents()
+	for(var/obj/item/gun as anything in typesof(/obj/item/gun/ballistic/shotgun))
```

Comment:
Make a `valid_typesof` proc (we only have `valid_subtypesof` right now 😢 ), then replace the `typesof` call with it.
```suggestion
	for(var/obj/item/gun as anything in typesof(/obj/item/gun/ballistic/shotgun))
```
Same for all other typesof call in this file.